pages/2_subscriptions_report.py

```diff
@@ -16,34 +17,113 @@
+subscription_data = data[
+    (data['billing_type'].isin(['subscription', 'recurring'])) & 
+    (data['transaction_type'] == 'sale')
+]
+
+# Filter the Dataframe to include only 'one-time' and 'invoiceitem' billing types
+onetime_data = data[
+    (data['billing_type'].isin(['one-time', 'invoiceitem'])) & 
+    (data['transaction_type'] == 'sale')
+]
+
+# Group by 'payment_month' to calculate MRR and single order
+mrr_data = subscription_data.groupby(subscription_data['payment_month'])['total_amount'].sum().sort_index()
+single_order_data = onetime_data.groupby(onetime_data['payment_month'])['total_amount'].sum().sort_index()
+
 ## KPI Metrics which need to be updated.
 with st.container():
     col1, col2, col3, col4, col5 = st.columns(5)
 
+    # Total Revenue From Subscriptions
     with col1:
-        st.metric(label="**Total Revenue From Subscriptions**", value="TBD")
+        current_total_revenue = subscription_data['total_amount'].sum()
+        last_year_total_revenue = subscription_data[subscription_data['payment_month'].dt.year == (subscription_data['payment_month'].dt.year.max() - 1)]['total_amount'].sum()
+        last_year_total_revenue_str = f'${current_total_revenue:,.2f}' if not pd.isna(current_total_revenue) else "no data"
+        yoy_total_revenue = current_total_revenue - last_year_total_revenue
+        yoy_total_revenue_str = f'{yoy_total_revenue:,.2f} YoY' if not pd.isna(yoy_total_revenue) else "no data"
+        st.metric(
+            label="**Total Revenue From Subscriptions**", 
+            value=last_year_total_revenue_str,
+            delta=yoy_total_revenue_str  # Use conditional string
+        )
 
+    # Active Subscriptions
     with col2:
-        st.metric(label="**Active Subscriptions**", value="TBD")
+        current_active_subscriptions = subscription_data[subscription_data['subscription_status'] == 'active']['subscription_status'].count()
+        current_active_subscriptions_str = f'{current_active_subscriptions}' if not pd.isna(current_active_subscriptions) else "no data"
+        last_year_active_subscriptions = subscription_data[
+            (subscription_data['subscription_status'] == 'active') & 
+            (subscription_data['payment_month'].dt.year == (subscription_data['payment_month'].dt.year.max() - 1))
+        ]['subscription_status'].count()
+        yoy_active_subscriptions = current_active_subscriptions - last_year_active_subscriptions
+        yoy_active_subscriptions_str = f'{yoy_active_subscriptions:,.0f} YoY' if not pd.isna(yoy_active_subscriptions) else "no data"
+        st.metric(
+            label="**Active Subscriptions**", 
+            value=current_active_subscriptions_str,
+            delta=yoy_active_subscriptions_str  # Use conditional string
+        )
 
+    # New Subscriptions
     with col3:
-        st.metric(label="**New Subscriptions**", value="TBD")
+        current_new_subscriptions = subscription_data['subscription_status'].count()
```

Comment:
It is a bit confusing that we can have a scenario where there are more New Subscriptions than Active Subscriptions in our sample data.

![image](https://github.com/user-attachments/assets/0f9dbe21-f824-4b4a-9077-c858912e4a1d)

Should we consider adjusting this code to also only look at `active` subscriptions, but are new within the period?
```suggestion
        current_new_subscriptions = subscription_data['subscription_status'] == 'active']['subscription_status'].count()
```

Reply:
Open to your thoughts here as this will impact downstream charts as well.

Reply:
I was wondering about that, but since the date range is about a year, I thought it would make sense that all new subscriptions wouldn't necessarily still be active after a year. I think that could tell a story, for example, that only half the people that start a subscription still have it active by the current period. But, I also see how that would be confusing for a demo and the benefit of keeping it limited to active to tell the story of the active subscriptions, how much are new. In that case, I have updated with your suggestion.

Reply:
Thanks @fivetran-catfritz I agree and think this is something we can definitely expand on once we move past the demo and into deployment. Thanks for making the requested update!